research/global-config/index.md

```diff
@@ -0,0 +1,361 @@
+upstream_project_url: -OVERRIDE ME-
+
+upstream_tag_template: v{version}
+
+jobs:
+  - job: pull_from_upstream
+    trigger: release
+    dist_git_branches:
+      - fedora-rawhide
+  - job: koji_build
+    trigger: commit
+    allowed_pr_authors: ["packit"]
+    dist_git_branches:
+      - fedora-rawhide
+```
+
+#### https://gitlab.gnome.org/packit/templates/configs/default_packit.yaml
+
+```yml
+# Gnome default packit config
+inherit: https://github.com/packit/templates/configs/standard-pull-from-upstream.yaml
+
+jobs:
+  - job: koji_build
+    allowed_pr_authors: ["packit", "gnome-admins"]
+
+  - job: tests
+    trigger: pull_request
+    tmt_plan: "smoke|full|packit-integration"
+    targets:
+      - fedora-rawhide
+```
+
+#### https://gitlab.gnome.org/package/packit.yaml
+
+```yml
+# A gnome package packit config
+inherit: https://gitlab.gnome.org/packit/templates/configs/default_packit.yaml
+
+specfile_path: specfile_path
+upstream_package_name: upstream_package_name
+downstream_package_name: downstream_package_name
+upstream_project_url: upstream_package_url
+
+jobs:
+  - job: tests
+    packages: ["downstream_package_name"]
+    tmt_plan: "smoke|full|packit-integration|package-tests"
+```
+
+### PROs and CONs
+
+#### Templating
+
+##### Pros
+
+Flexible, probably the most flexible implementation that allows to freely mix configuration snippets for creating a customized final configuration.
+
+##### Cons
+
+The "pure" templating mechanism, in the above example, requires the package maintainer to know that koji builds, in the gnome ecosystem, should be allowed for any **gnome-admin**, instead _inheritance_ and _global config + overlays_ encapsulate well the knowledge in the middle layer packit config.
+
+Templating is flexible but on the other end it is more error prone; there is no _base configuration_ and a packager can list templates in the wrong order.
+
+Probably, in the end, the packager will use smaller config snippets, decreasing performance and readability.
+
+#### Global config + overlays
+
+##### Pros
+
+Good knowledge encapsulation in middle layers (see the **gnome-admin** for allowed_pr_authors in the above example).
+
+Explicit and thus easily readable, since the use of templating.
+
+Flexible, config snippets can easily be removed using template conditional functionalities (as in the above example for the test job with trigger commit).
+
+##### Cons
+
+The templating syntax can be more error prone if compared with inheritance.
+
+#### Inheritance
+
+##### Pros
+
+Concise, it's the most concise syntax we could use and probably the least error prone.
+
+##### Cons
+
+Poor flexibility, I don't see an easy way to disable the above test job with trigger commit.
+
+Not really explicit, even though we use a placeholder it is harder to recognize the keys that need overriding.
+
+### Implementation
+
+Personally I find the _global config + overlays_ approach the best and in this case we would need to:
```

Comment:
:+1: Looking at the examples (thanks for them), this one looks the most readable.